### ISSUE TITLE
Notification Mail body uses wrong host name

### ISSUE THREAD

**Issue by ronator** [Icinga/icinga2#5473]
Since the upgrade to 2.7 the layout of notification mails have changed. The mail body confusingly uses the icinga host name in the "Host:" field instead of the affected server.

## Expected Behavior

The mail should look like this:

```
***** Icinga 2 Service Monitoring on icingaweb2de *****

==> Mail Queue Status on Ansible is OK! <==

Info:    OK: postfix mailq reports queue is empty

When:    2017-08-08 11:00:52 +0200
Service: Mail Queue Status (Display Name: "Mail Queue Status")
Host:    ansible.somedomain.local (Display Name: "Ansible")
IPv4:    192.168.xxx.xxx
```

## Current Behavior

Instead, it show the icinga host name which is misleading:

```
***** Icinga 2 Service Monitoring on icingaweb2de *****

==> Mail Queue Status on Ansible is OK! <==

Info:    OK: postfix mailq reports queue is empty

When:    2017-08-08 11:00:52 +0200
Service: Mail Queue Status (Display Name: "Mail Queue Status")
Host:    icingaweb2de (Display Name: "Ansible")
IPv4:    192.168.xxx.xxx
```

Downtime mails are also affected:

```
***** Icinga 2 Host Monitoring on icingaweb2de *****

==> DE vCenter Live (icingaweb2de) is UP! <==

Info:    PING OK - Packet loss = 0%, RTA = 0.24 ms

When:    2017-08-07 15:05:28 +0200
Host:    icingaweb2de (Display Name: "DE vCenter Live")
IPv4:    192.168.30.120

Comment by another.admin:   Update
```
The icinga hostname is used instead ot the real hostname of the affected host.

## Possible Solution

Correct usage of variables in mail-service-notification.sh, I guess. Did not investigate further but seems related to issue #5472

## Steps to Reproduce (for bugs)

1. upgrade to 2.7
2. provoke a service or host mail
3. check the mail layout

## Context

It makes it a little harder to read the mail and I think this is not the expected behaviour.

## Your Environment

* Version used (`icinga2 --version`):

icinga2 - The Icinga 2 network monitoring daemon (version: r2.7.0-1)

Copyright (c) 2012-2017 Icinga Development Team (https://www.icinga.com/)
License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl2.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Application information:
  Installation root: /usr
  Sysconf directory: /etc
  Run directory: /run
  Local state directory: /var
  Package data directory: /usr/share/icinga2
  State path: /var/lib/icinga2/icinga2.state
  Modified attributes path: /var/lib/icinga2/modified-attributes.conf
  Objects path: /var/cache/icinga2/icinga2.debug
  Vars path: /var/cache/icinga2/icinga2.vars
  PID path: /run/icinga2/icinga2.pid

* Operating System and version:

System information:
  Platform: Ubuntu
  Platform version: 14.04.5 LTS, Trusty Tahr
  Kernel: Linux
  Kernel version: 4.4.0-89-generic
  Architecture: x86_64

Build information:
  Compiler: GNU 4.8.2
  Build host: e1e9a326f1b8

* Enabled features (`icinga2 feature list`):

Disabled features: debuglog gelf graphite influxdb opentsdb syslog
Enabled features: api checker command compatlog ido-mysql livestatus mainlog notification perfdata statusdata

* Icinga Web 2 version and modules (System - About):

Version 2.4.1
businessprocess | 2.1.0
doc | 2.4.1
generictts | 2.0.0
monitoring | 2.4.1
nagvis | 1.0.0
pnp | 1.0.1

* Config validation (`icinga2 daemon -C`):

information/cli: Icinga application loader (version: r2.7.0-1)
information/cli: Loading configuration file(s).
information/ConfigItem: Committing config item(s).
information/ApiListener: My API identity: IcingaWeb2
information/ConfigItem: Instantiated 4 ApiUsers.
information/ConfigItem: Instantiated 1 ApiListener.
information/ConfigItem: Instantiated 5 Zones.
information/ConfigItem: Instantiated 1 FileLogger.
information/ConfigItem: Instantiated 3 Endpoints.
information/ConfigItem: Instantiated 1 LivestatusListener.
information/ConfigItem: Instantiated 2 NotificationCommands.
information/ConfigItem: Instantiated 3445 Notifications.
information/ConfigItem: Instantiated 225 CheckCommands.
information/ConfigItem: Instantiated 72 Downtimes.
information/ConfigItem: Instantiated 269 Hosts.
information/ConfigItem: Instantiated 1 IcingaApplication.
information/ConfigItem: Instantiated 48 HostGroups.
information/ConfigItem: Instantiated 15 Comments.
information/ConfigItem: Instantiated 1158 Dependencies.
information/ConfigItem: Instantiated 8 UserGroups.
information/ConfigItem: Instantiated 4 TimePeriods.
information/ConfigItem: Instantiated 12 Users.
information/ConfigItem: Instantiated 2203 Services.
information/ConfigItem: Instantiated 30 ServiceGroups.
information/ConfigItem: Instantiated 63 ScheduledDowntimes.
information/ConfigItem: Instantiated 1 CompatLogger.
information/ConfigItem: Instantiated 1 StatusDataWriter.
information/ConfigItem: Instantiated 1 ExternalCommandListener.
information/ConfigItem: Instantiated 1 CheckerComponent.
information/ConfigItem: Instantiated 1 IdoMysqlConnection.
information/ConfigItem: Instantiated 1 NotificationComponent.
information/ConfigItem: Instantiated 1 PerfdataWriter.
information/ScriptGlobal: Dumping variables to file '/var/cache/icinga2/icinga2.vars'
information/cli: Finished validating the configuration file(s).

**Comment by dnsmichi** [comment]
Did you update your NotificationCommand objects too? Presumingly not, and as such the wrong parameters might be set.

**Comment by ronator** [comment]
No, I did not. Was it mentioned in the change log? I mean besides the blog entry on "making icinga more stable with 2.7". Anyway, how would I do that? And in what way? The only config changes the upgrade provided were related to conf.d/ and not to zones.d/ so that change was not expected at least by me.
Thank you for explaining the situation.

**Comment by dnsmichi** [comment]
Rough tone.

See the referenced PR which adds some more details on required configuration changes and should probably resolve your issue as well.

**Comment by ronator** [comment]
You consider this rough? That wasn't my intention.

This change was not clear to me, the blog entry was edited today to reflect that now in more detail, so thanks. I am very sorry that I understood the title "we focused on stability" as "we focused on stability". I cannot remember that the passage about notifications in the blog entry was there when 2.7 was released.

Afterwards, I was wondering about the deployment style - that I have to change the configs myself whereas lots of configs are provided by the package installation. Now I have to read, maybe this will explain to me why I would have to make changes manually. Thanks again for pointing out how to fix what I obviously have missed.

I guess what @dnsmichi is trying to say is that this issue can be closed, although I already found others who ran into the same problem - maybe we all should read more carefully.

**Comment by dnsmichi** [comment]
Your tone and sarcasm did not really change to the better in the past years, so I am leaving this to you to fix it accordingly. Hint: $HOSTNAME might evaluate to something different if not explicitly set by the daemon.

**Comment by ronator** [comment]
I can't remember discussing with you that much, especially since my team and I did experience some bewildering arrogance @icingacamp in Berlin by yourself.

I really can't follow your statement. I can't tell why you insert any sarcasm that is not there - I was truly asking how I should have been aware of that and where you spread the word.

I was asking some questions here in return to your answer because I was blamed that I did not do the appropriate steps while the needed infos were missing in the blog entry when released regarding cluster setups. I basically asked, from where I should have gotten these info beforehand. If there had been complete info on that, this issue would have not appeared. Surely, I only asked to prevent such mistakes by myself.

But I was totally denied this info, maybe because we all knew it wasn't out there yet, but then a small change to the blog entry was added afterwards. The blog entry still seems to be updated a lot. 

Furthermore, your first answer was not really helping at all because it was willingly kept quite general, which may be caused by your opinion on my rough tone - so you never really wanted to answer to me - or maybe it also was that arrogant tone of yours we are referring to that increased over the last years.

As you tweeted, "sarcasm may not help in solving problems" - keep in mind, that this also applies to arrogant answer and statements willingly making other people feel bad or stupid.

Maybe I was misunderstood and it was taken personally when I referred to the blog title "for stability and performance".

**Comment by dnsmichi** [comment]
I was wrong and I hereby apologise. 

https://www.icinga.com/docs/icinga2/latest/doc/03-monitoring-basics/#notification-commands-in-27 is fixed for an upcoming v2.7.1 and should help resolve your issue. You won't need any changes to your notification apply rules and templates, unless you want to explicitly enable things like verbose logging, or set an additional URL for Icinga Web 2 (those are new features which did not exist in the old script).

https://github.com/Icinga/icinga2/blob/master/etc/icinga2/conf.d/templates.conf#L76
https://github.com/Icinga/icinga2/blob/master/etc/icinga2/conf.d/notifications.conf#L18
